File: frontend-vue/src/services/admin/AdminOptionService.js
import http from "@/utils/http-common";
import LoginHeader from "../login/LoginHeader";

class AdminOptionService {
    // 전체조회
    getAll(opName, page, size) {
        return http.get(`/admin/admin-option?opName=${opName}&page=${page}&size=${size}`,{
            headers: LoginHeader()
        });
    }
    // 상세조회
    get(opId) {
        return http.get(`/admin/admin-option/${opId}`,{
            headers: LoginHeader()
        });
    }
    // 상품등록함수
    create(data) {
        return http.post("/admin/admin-option", data,{
            headers: LoginHeader()
        });
    }
    // 상품수정함수
    update(opId, data) {
        return http.put(`/admin/admin-option/${opId}`, data,{
            headers: LoginHeader()
        });
    }
    // 상품삭제함수
    delete(opId) {
        return http.delete(`/admin/admin-option/deletion/${opId}`,{
            headers: LoginHeader()
        });
    }
}

export default new AdminOptionService();

File: 01_Backend/backend/src/main/java/org/example/backend/service/admin/AdminOptionService.java
package org.example.backend.service.admin;

import org.example.backend.model.entity.admin.AdminOption;
import org.example.backend.repository.admin.AdminOptionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class AdminOptionService {
    @Autowired
    AdminOptionRepository adminOptionRepository;

    //    TODO: 전체조회(read)
    public List<AdminOption> findAll() {
        List<AdminOption> list = adminOptionRepository.findAll();
        return list;
    }

    //    페이징 처리
    public Page<AdminOption> findAllByAdminOptionNameContaining(String opName,
                                                                Pageable pageable) {
        Page<AdminOption> page
                = adminOptionRepository
                .findAllByAdminOptionNameContaining(opName, pageable);
        return page;
    }

    //    상세조회
    public Optional<AdminOption> findById(Long opId) {
        Optional<AdminOption> optionalAdminOption
                = adminOptionRepository.findById(opId);
        return optionalAdminOption;
    }

    //    TODO: 등록(insert),수정(update)
    public AdminOption save(AdminOption adminOption) {
        AdminOption adminOption2 = adminOptionRepository.save(adminOption);

        return adminOption2;
    }

    //    TODO: 삭제(delete)
    public boolean removeById(Long opId) {
        if (adminOptionRepository.existsById(opId) == true) {
            adminOptionRepository.deleteById(opId);
            return true;
        } else {
            return false;
        }
    }
}
